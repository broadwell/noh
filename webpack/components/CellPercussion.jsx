import React from "react";
import PropTypes from "prop-types";

const CellPercussion = props => (
  <div className={`cell cell--${props.length}`}>{props.text}</div>
);

CellPercussion.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellPercussion;
