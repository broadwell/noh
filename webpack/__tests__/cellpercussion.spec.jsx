import React from "react";
import { shallow } from "enzyme";
import CellPercussion from "../components/CellPercussion";

describe("<CellPercussion>", () => {
  it("renders as expected by default", () => {
    const component = shallow(<CellPercussion text="mitsuji" length={14} />);
    expect(component).toMatchSnapshot();
  });
});
