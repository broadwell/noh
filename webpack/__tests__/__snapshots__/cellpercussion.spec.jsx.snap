// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellPercussion> renders as expected by default 1`] = `
<div
  className="cell cell--14"
>
  mitsuji
</div>
`;
